### PR TITLE
ESP32S3 temperature sensor 

### PR DESCRIPTION
**Pull request by wmrsouza**
ESP32S3 internal temperature sensor supported

### CODE REVIEW

**Review by LucasTambor — DISMISSED** [review]

**Review by sylvioalves — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]
Tested on M5Stack AtomS3 :)

Note: do the other esp32 boards in the sample really need `CONFIG_REQUIRES_FULL_LIBC=y` btw?

File: samples/sensor/die_temp_polling/boards/esp32s3_devkitm.overlay [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2023 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Application overlay for creating temperature sensor device instance
```
Comment (kartben) [inline samples/sensor/die_temp_polling/boards/esp32s3_devkitm.overlay:6]: Actually, a nit (which I realize should be probably be fixed in other ESP32 overlays as well): 

```suggestion
 * Application overlay for enabling temperature sensor device instance
```

**Review by MaureenHelm — DISMISSED** [review]

File: dts/bindings/sensor/espressif,esp32-temp.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ properties:
   range:
     type: int
     description: |
-      The temperature sensor is available on the ESP32-S2, ESP32-C3. Note
+      The temperature sensor is available on the ESP32-S2, ESP32-S3, ESP32-C3. Note
```
Comment (marekmatej) [inline dts/bindings/sensor/espressif,esp32-temp.yaml:14]: please move Note at the new line

**Review by marekmatej — DISMISSED** [review]
some nit-picks otherwise lgtm

**Review by kartben — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**